active/AnimatedIcon/AnimatedIconSpec.md

```diff
@@ -0,0 +1,282 @@
+
+# Animated Icon
+
+## Background
+
+Xaml has an
+[IconElement](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.IconElement)
+base type today with derived types like
+[BitmapIcon](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.BitmapIcon)
+and [SymbolIcon](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.SymbolIcon).
+Since it's an element it can be used in the UI tree, such as in a 
+[StackPanel](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.StackPanel).
+
+All of these icons are static though, and developers do not have a straightforward way to
+add an animated icon in controls where they can use a static icon.
+There is also a desire to have an animated icon respond to user interactivity via state changes to either draw
+attention to an action a user can take or to add user delight.
+
+The [AnimatedVisualPlayer](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.AnimatedVisualPlayer)
+element can be used to create animated (Lottie-based) icons,
+but to have the animation be driven by the states of and interactions with a control,
+developers may need to write a lot of custom code.
+
+Also, AnimatedVisualPlayer does not inherit from IconElement so it cannot be used in places that
+need one, like 
+[NavigationViewItem.Icon](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.NavigationViewItem.Icon)
+or in a button.
+
+This spec introduces an AnimatedIcon IconElement which reduces, or ideally eliminates, the amount of code-behind needed to
+implement most scenarios, and it can be used in WinUI controls that support an IconElement.
+
+# How to design animations for AnimatedIcon
+
+## Creating animated content with Lottie 
+
+1. Download the Lottie file for the icon you are looking to add to your application.
+2. Run the Lottie file through LottieGen. The LottieGen will output the file that you will add to your project that will include the markers needed for animated icon to map the correct state transitions to the correct animation segments. It will also contain the themable color properties you can set in code. 
+3. Take the output file from LottieGen and place it in your project.
+
+## Design Requirements  
+
+Designers will need to add markers named like the following to all Lottie files that will be used with AnimatedIcon.
+The marker names should reflect the visual state the animation is going from to the visual state the animation is
+going to. For example, the bare minimum of the markers that are needed for icons being used in a Button are:  
```

Comment:
How do you know for each control what the list of special marker names are?

Reply:
That's a great question. The plan is to update the icon documentation to talk about Animated icons and to also create guidance specifically for animated icon to talk about the formatting and when/where to use animated icons.

Reply:
Shouldn't the documentation be on each control? Different controls will have different states.

Reply:
I think that the (small) majority of animated icon use cases will only respond to interaction events (PointerOver/pressed/normal) states so I think this basic use case can be documented with animated Icon.  Controls which have more specific needs (ie checkbox) should document the state names in their control documentation.

Reply:
Agreed, and more specicially we should list which control's default template support animated icons with what markers. Its not the control which does the work to support AnimatedIcon its the template.

Reply:
Updated.